server/metadata/tenant.go

```diff
@@ -1536,31 +1536,71 @@ func (tenant *Tenant) String() string {
 	return fmt.Sprintf("id: %d, name: %s", tenant.namespace.Id(), tenant.namespace.StrId())
 }
 
-// Size returns approximate data size on disk for all the collections, databases for this tenant.
-func (tenant *Tenant) Size(ctx context.Context) (int64, error) {
+func (tenant *Tenant) listEncCollName(db *Database) [][]byte {
+	colls := make([][]byte, 0, len(db.collections))
+	for _, v := range db.collections {
+		name, _ := tenant.Encoder.EncodeTableName(tenant.namespace, db, v.collection)
+		colls = append(colls, name)
+	}
+
+	return colls
+}
+
+func (tenant *Tenant) sumCollsSize(ctx context.Context, colls [][]byte) (*kv.TableStats, error) {
+	var sumStats kv.TableStats
+
+	for _, coll := range colls {
+		stats, err := tenant.kvStore.GetTableStats(ctx, coll)
+		if err != nil {
+			return nil, err
+		}
+
+		sumStats.StoredBytes += stats.StoredBytes
+		sumStats.OnDiskSize += stats.OnDiskSize
+		sumStats.RowCount += stats.RowCount
+	}
+
+	return &sumStats, nil
+}
+
+// Size returns exact data size on disk for all the collections, databases for this tenant.
+func (tenant *Tenant) Size(ctx context.Context) (*kv.TableStats, error) {
```

Comment:
If we are not counting branch database size, let's add a note here, and we can incrementally add it.

Reply:
Include branches now. Good catch.